### PR TITLE
Advanced bot - Minimum features | Almost everything changed

### PR DESCRIPTION
**Pull request by TheDogHusky**
——————
Liberty Bot
——————
> What’s new?

> New commands, new categories. Linked to mongodb and customizable prefix setup.
> Handler changed, and corrected some bugs.
> Logger linked to client (client.logger)

### CODE REVIEW

File: src/commands/Info/clientinfo.js [hunk 1]
```diff
@@ -6,9 +6,9 @@ const { MessageEmbed } = require("discord.js");
 module.exports = {
 	name: "clientinfo",
 	desc: "Get clientinfo",
-	cat: "Info",
+	category: "Info",
 	usage: "clientinfo",
-	async command(bot, message, args, extras) {
+	async command(bot, message, args, extras, data) {
```
Comment (GamearooCodes) [inline src/commands/Info/clientinfo.js:11]: data can you explain what its used for and why it cant be part of extras?

File: src/commands/Info/ping.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            embed.setColor('ffb200');
+        } else if(client.ws.ping >= 600) {
+            apiPingCommentaire = 'Very Bad';
+            embed.setColor('ff0000');
+        };
+            
+        if(botPing < 300) {
+            botPingCommentaire = 'Stable';
+            embed.setColor('00ff2a');
+        } else if(botPing >= 300) {
```
Comment (GamearooCodes) [inline src/commands/Info/ping.js:31]: remove bad to us stable is 600 and below and lag is 600 and above

File: src/commands/Info/ping.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        let apiPingCommentaire;
+        const embed = new MessageEmbed();
+        embed.setTitle('**🏓Pong !🏓**');
+            
+            
+        if(client.ws.ping < 300) {
+            apiPingCommentaire = 'Stable';
+            embed.setColor('00ff2a');
+        } else if(client.ws.ping >= 300) {
+            apiPingCommentaire = 'Bad';
```
Comment (GamearooCodes) [inline src/commands/Info/ping.js:21]: remove bad to us stable is 600 and below and lag is 600 and above

File: src/commands/Mod/ban.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        const member = message.mentions.members.first()
+        if (!member) return message.reply({ content: `Please ping the user to ban.`, allowedMentions: { repliedUser: false } })
+        if (member.id === message.guild.ownerId) return message.reply({ content: `You can't ban the server owner!`, allowedMentions: { repliedUser: false } })
+        if (message.member.roles.highest.comparePositionTo(member.roles.highest) < 1 && message.author.id !== message.guild.ownerId) return message.reply({ content: `I'm sorry but this user is superior to me.`, allowedMentions: { repliedUser: false } })
+        if (member.permissions.has(Permissions.FLAGS.BAN_MEMBERS)) return message.reply({ content: `I'm sorry but this user have the same permissions as me, I can't ban him.`, allowedMentions: { repliedUser: false } })
+        const reason = args.slice(2).join(' ') || 'Reason not specified'
+        await member.ban({reason})
+        message.reply({ content: `${member.user.tag} has been banned for reason: \`${reason}\``, allowedMentions: { repliedUser: false } })
```
Comment (GamearooCodes) [inline src/commands/Mod/ban.js:18]: add a logs system to a channel if you can

File: src/commands/Utils/avatar.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+	name: "avatar",
+	desc: "Gets user's avatar!",
+	category: "Utils",
+	usage: "avatar [user]",
+	async command(client, message, args, extras, data) {
+        const optionnalMember = message.mentions.members.first();
```
Comment (GamearooCodes) [inline src/commands/Utils/avatar.js:7]: please remove this command we do not collect or send data that a user can abuse to do illegal or breaking tos with

File: src/events/msg.js [hunk 6]
```diff
@@ -3,19 +3,23 @@ const { run } = require("./ready");
...
+		const schema = require('../../models/guild')
+        const data = await schema.findOne({ GuildID: message.guild.id })
+        if(!data) return message.reply({ content: `There is no schemas existing for the guild. Creating one, please wait...`, allowedMentions: { repliedUser: false } }) && new schema({ GuildID: message.guild.id }).save().then(() => message.reply({ content: `The schema has been created. You can now use the commands.`, allowedMentions: { repliedUser: false } }));
+		let prefix = data.Prefix;
 		if (!message.content.toLowerCase().startsWith(prefix)) return;
-
 		let args = message.content.substring(prefix.length).split(" ");
 
 		const extras = {
 			hello: "Hello",
```
Comment (GamearooCodes) [inline src/events/msg.js:16]: if you can move data to the extras to pass it into commands ty

**Review by GamearooCodes — CHANGES_REQUESTED** [review]
Fix any problems i mentioned in the review

File: src/commands/Info/clientinfo.js (hunk 1 above)
Comment (TheDogHusky) [reply src/commands/Info/clientinfo.js:11]: Data represents the GuildSchema. Yeah I can import it in extras

File: src/commands/Info/ping.js (hunk 3 above)
Comment (TheDogHusky) [reply src/commands/Info/ping.js:21]: I’ll change it webnesday

File: src/commands/Mod/ban.js (hunk 4 above)
Comment (TheDogHusky) [reply src/commands/Mod/ban.js:18]: Just a channel send to a special property in the schema? Like I do a log system? Okay I’ll do that too

File: src/commands/Utils/avatar.js (hunk 5 above)
Comment (TheDogHusky) [reply src/commands/Utils/avatar.js:7]: K

File: src/events/msg.js (hunk 6 above)
Comment (TheDogHusky) [reply src/events/msg.js:16]: Okay I’ll change it

File: src/commands/Mod/ban.js (hunk 4 above)
Comment (GamearooCodes) [reply src/commands/Mod/ban.js:18]: yes a embed of the log of who banned it and not maybe make a setting to enable/disable the log

Comment (TheDogHusky) [reply src/commands/Mod/ban.js:18]: maybe a setting just to enable or disable the mod logs

**Comment by TheDogHusky** [comment]
The log system has been added, I let you review the pull request!

File: package.json [hunk 7]
```diff
@@ -21,8 +21,11 @@
 		"discord.js": "^13.3.1",
 		"moment": "^2.29.1",
 		"mongoose": "^5.13.7",
+		"npm": "^8.3.0",
+		"ram-api.js": "^5.0.0",
 		"systeminformation": "^5.9.17",
 		"winston": "^3.3.3",
-		"winston-console-format": "^1.0.8"
+		"winston-console-format": "^1.0.8",
+		"yarn": "^1.22.17"
```
Comment (GamearooCodes) [inline package.json:29]: please remove yarn and use npm i -g yarn or change the package manager to npm

**Review by GamearooCodes — CHANGES_REQUESTED** [review]
please fix the package manager issues

File: package.json (hunk 7 above)
Comment (TheDogHusky) [reply package.json:29]: why? Yarn is already installed and works perfectly

Comment (TheDogHusky) [reply package.json:29]: image
it's already fixed btw

Comment (TheDogHusky) [reply package.json:29]: there is no packager issue idk why it's like that on github

**Pull request merged**